lib/guest.ts

```diff
@@ -3,20 +3,66 @@
  * SPDX-License-Identifier: GPL-3.0-or-later
  */
 import { getBuilder } from '@nextcloud/browser-storage'
+import { TypedEventTarget } from 'typescript-event-target'
 
 const browserStorage = getBuilder('public').persist().build()
 
+/**
+ * This event is emitted when the list of registered views is changed
+ */
+interface UpdateGuestDisplayName extends CustomEvent<never> {
+	type: 'updateDisplayName'
+}
+
+class GuestUser extends TypedEventTarget<{ updateDisplayName: UpdateGuestDisplayName }> {
```

Comment:
Maybe also add the inheritance of `NextcloudUser` if you already implement `isAdmin`?

```suggestion
class GuestUser extends TypedEventTarget<{ updateDisplayName: UpdateGuestDisplayName }> implements NextcloudUser {
```